ngi_reports/stockholm/project_summary.py

```diff
@@ -199,3 +229,55 @@ def get_order_dates(self):
         if self.creation_date:
             dates.append("_Report date:_ {}".format(self.creation_date))
         return ", ".join(dates)
+
+
+    def get_library_method(self):
+        """Get the library construction method and return as formatted string
+        """
+        if self.proj.get('application') == "Finished library":
+            return "Library was prepared by user."
+        try:
+            lib_head = ['Input', 'Type', 'Option', 'Category']
+            lib_meth = re.match(r'(.*) \[[0-9]+\]',self.proj_details['library_construction_method'])
+            if lib_meth:
+                return ("\n".join(["* {}: {}".format(lib_head[c],i.strip()) for c,i in enumerate(lib_meth.group(1).split(',')) \
```

Comment:
Wow, you're really going for it with these one-liners! :grin: 

```python
lib_meth = self.proj_details['library_construction_method'].rstrip('1234567890')
lib_meths = lib_meth.split(',')
lib_preps = []
for c, i in enumerate(lib_meths):
    category = lib_head[c]
    thismeth = i.strip()
    if thismeth[:1] != '-':
        lib_preps.append("* {}: {}".format(category, thismeth))
return ( "\n".join(lib_preps))
```

For the first regex, I'm not quite sure what it's meant to be doing? Removing numbers from the end of the string? It doesn't have any anchors (`^` or `$`), so the `.*` could do weird things. I swapped it for something a little clearer.

For the second regex (`^[^-\s]`) - the second part isn't needed I think, as the string has already been passed though `strip()`, which means that you're only checking that the first character isn't a hyphen.